In [1]:
# find an area of a triangle using python.
base = 10
height = 7
area = (1/2) * base * height
area

35.0

In [2]:
type(area)

float

<h3 style="color:purple">Two type of numbers: Integers and Float</h3>

In [3]:
weight = 100.5
type(weight)

float

In [4]:
family_members = 5
type(family_members)

int

<h3 style="color:purple">Basic airthmetic with numbers</h3>

In [6]:
1+2

3

In [7]:
10/3

3.3333333333333335

In [8]:
# floor division - gives only int o/p
10//3

3

In [9]:
# modulo operator - gives the reminder
17%3

2

In [10]:
3*2

6

In [12]:
# power, square
3**2

9

In [13]:
10+2*3

16

In [14]:
(10+2)*3

36

In [15]:
type(5)

int

In [16]:
type(5.2)

float

In [18]:
a = 2.3e4
a

23000.0

In [19]:
1e4

10000.0

In [20]:
1e-4

0.0001

In [21]:
food=120.34
rent=500.67
utilities=200.35
total_expense = food + rent + utilities
total_expense

821.36

In [22]:
round(total_expense)

821

In [24]:
# number value dont have quotes. if quotes, it is considered as string
food = "120.34"
rent = "500.67"

total = food + rent
total
#string concat

'120.34500.67'

In [25]:
type(total)

str

In [26]:
type(total_expense)

float

<h3 style="color:purple">Binary Representation Of Integers</h3>

In [28]:
format(5, 'b')

'101'

In [29]:
# 1*2^2 + 0*2^1+ 1*2^0 = 4+0+1 = 5

<h2 style="color:purple">Excercise: </h2>

1. Pass 145 and ‘o’ in the format() function and see what it returns. Try to find out which representation it is.
2. 
There is a circular pond in a village. This pond has a radius of 84 meters. Can you find the area of the pond? (Bonus: If there is exactly 1.4 liter of water in a square meter, what is the total amount of water in the pond?)
Hin :
Circle Area = 
3. 2

If you cross a 490-meter-long street in 7 minutes, then what is your speed in meters per second? Print your answer without any decimal point in it.

In [31]:
# format function to formate the number
format(145, 'o')

'221'

In [33]:
# 2*8^2 + 2*8^1 + 1*8^0 = 2*64+ 2*8 +1*1 = 128+ 16+ 1 = 145
#'o' stands for `octal representation` of the number.

In [37]:
#2. circular pond
radius = 84
pond_area = 3.14 * radius ** 2
print(pond_area)

22155.84


**BONUS**

Because there is exactly 1.4 liter of water in 1 square meter. And we know that the total area of pond is `22155.84` square meter.

We can get the total amount of water if we multiply the given amount wih given area. 

In [35]:
per_sqt_meter_water = 1.4
total_water = pond_area * per_sqt_meter_water
print(total_water)


31018.176


In [36]:
#There is total `31018.176` liter water in the pond.

In [39]:
#3.
distance = 490
time = 7 * 60 # convert minute to seconds
time

420

In [41]:
speed = distance / time
print(speed)

1.1666666666666667


In [42]:
print(round(speed))

1
